libs/logger.js

```diff
@@ -0,0 +1,22 @@
+import winston from 'winston';
+
+/**
+ * Log function that outputs console logs in json format.
+ *
+ * @param {('error'|'warn'|'info'|'http'|'verbose'|'debug'|'silly')} level Log level based on npm log level standard.
+ * @param {string} message   Log message.
+ * @param {string} requestId Request id to be able to filter logs on a single request.
+ * @param {string} errorCode Error code for locating log in code.
+ * @param {object} data      Custom data related to log.
+ * @returns {void}
+ */
+export const logger = (level, message, requestId, errorCode, data = {}) => {
+  const logger = winston.createLogger({
+    level: process.env.LOG_LEVEL || 'info',
+    format: winston.format.json(),
+    transports: [new winston.transports.Console()],
+    exitOnError: false,
+  });
```

Comment:
Perhaps a small optimization in practice but could we cache this somehow for the duration of the lambda execution? This is (re)initialized for every log message.

Reply:
Agree, I could just move it outside the function like this.

Looks a bit unnessesary with a seperate funtion but my reasoning about having it as a seperate function and now just plain winston was to push devs to always fill in the info needed in monitoring.

```js
import winston from 'winston';

const logger = winston.createLogger({
  level: process.env.LOG_LEVEL || 'info',
  format: winston.format.json(),
  transports: [new winston.transports.Console()],
  exitOnError: false,
});

/**
 * Log function that outputs console logs in json format.
 *
 * @param {('error'|'warn'|'info'|'http'|'verbose'|'debug'|'silly')} level Log level based on npm log level standard.
 * @param {string} message   Log message.
 * @param {string} requestId Request id to be able to filter logs on a single request.
 * @param {string} errorCode Error code for locating log in code.
 * @param {object} data      Custom data related to log.
 * @returns {void}
 */
export const log = (level, message, requestId, errorCode, data = {}) => {
  logger.log(level, message, { errorCode: errorCode, requestId: requestId, data: data });
};
```